### PR TITLE
Restore 174

### PR DESCRIPTION
**Pull request by bananer**
Restore the changes in #212, adjusting for changes in #252

This reverts commit a1255785deb98db59e7cf9718d53ccc5917471eb.

### CODE REVIEW

**Comment by bananer** [comment]
@CyberShadow does this look ok to you?

File: src/build.sh [hunk 1]
```diff
@@ -303,7 +304,10 @@ for branch in ${BRANCH_NAME//,/ }; do
 
         DEBUG_LOG="$LOGS_DIR/$logsubdir/lineage-$los_ver-$builddate-$RELEASE_TYPE-$codename.log"
 
+        set +eu
+        # shellcheck source=/dev/null
```
Comment (CyberShadow) [inline src/build.sh:308]: `# shellcheck source=/dev/null` is not needed here. (It was used to tell ShellCheck to not attempt to load/parse the target of a `source` command.)

File: src/build.sh [hunk 2]
```diff
@@ -313,15 +317,19 @@ for branch in ${BRANCH_NAME//,/ }; do
         # Start the build
         echo ">> [$(date)] Starting build for $codename, $branch branch" | tee -a "$DEBUG_LOG"
         build_successful=false
+        set +eu
+        # shellcheck source=/dev/null
         if mka bacon &>> "$DEBUG_LOG"; then
+          set -eu
+
```
Comment (CyberShadow) [inline src/build.sh:324]: Is this change needed? I thought `mka` was a program and not a function, in which case it needs neither this nor the subshell wrapping (i.e. no `set +eu` at all).

In any case, `# shellcheck source=/dev/null` is also unnecessary here, as it does nothing to commands which are not `source`.

**Review by CyberShadow — APPROVED** [review]
I'm guessing that building succeeds with this; so, other than the above notes, LGTM, thank you.

File: src/build.sh (hunk 2 above)
Comment (bananer) [reply src/build.sh]: `mka` is actually another shell function: https://github.com/LineageOS/android_vendor_lineage/blob/7124e2cdbc66739ad80b160d222426ec5f9c2835/build/envsetup.sh#L719

running it without `set +e` failed for me, but running it in a subshell with it worked and seems like a better solution
